File: jvm/core/main/scala/com/webank/eggroll/core/deepspeed/job/meta/QueryJobRequest.scala
package com.webank.eggroll.core.deepspeed.job.meta

import com.google.protobuf.ByteString
import com.webank.eggroll.core.constant.StringConstants
import com.webank.eggroll.core.meta.Deepspeed

import scala.language.implicitConversions

case class QueryJobRequest(sessionId: String = StringConstants.EMPTY) {
}

object QueryJobRequest {
  implicit def serialize(src: QueryJobRequest): Array[Byte] = {
    Deepspeed.QueryJobRequest.newBuilder()
      .setSessionId(src.sessionId)
      .build()
      .toByteArray
  }

  implicit def deserialize(byteString: ByteString): QueryJobRequest = {
    val proto = Deepspeed.QueryJobRequest.parseFrom(byteString)
    QueryJobRequest(sessionId = proto.getSessionId)
  }
}

File: jvm/new_core/src/main/java/com/eggroll/core/pojo/QueryJobRequest.java
package com.eggroll.core.pojo;

import com.webank.eggroll.core.meta.Deepspeed;
import lombok.Data;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

@Data
public class QueryJobRequest implements RpcMessage{
    Logger log = LoggerFactory.getLogger(QueryJobRequest.class);
    private String sessionId;

    @Override
    public byte[] serialize() {
        Deepspeed.QueryJobRequest.Builder builder = Deepspeed.QueryJobRequest.newBuilder();
        builder.setSessionId(this.sessionId);
        return builder.build().toByteArray();
    }

    @Override
    public void deserialize(byte[] data) {
        try {
            Deepspeed.QueryJobRequest request = Deepspeed.QueryJobRequest.parseFrom(data);
            this.sessionId = request.getSessionId();
        } catch (Exception e) {
            log.error("deserialize error : ", e);
        }
    }
}
